tensorbay/cli/log.py

```diff
@@ -171,54 +224,83 @@ def _merge(self, latest_commit: Commit) -> bool:
         del self._keys[-1]
         return True
 
     def _sort(self) -> None:
         """Sort commits paging list by commit date."""
         # Only one branch exists.
-        if len(self._sorted_commits) == 1:
+        if len(self._sorted_logs) == 1:
             return
 
         # Binary insert.
-        commits = self._sorted_commits.pop()
+        log = self._sorted_logs.pop()
         del self._keys[-1]
-        date = commits[0].committer.date
+        date = log[0].committer.date
         index = bisect.bisect_left(self._keys, date)
-        self._sorted_commits.insert(index, commits)
+        self._sorted_logs.insert(index, log)
         self._keys.insert(index, date)
 
-    def _generate_commits(self) -> Iterator[Commit]:
-        """Get the latest commit in commit list.
+    def generate_commits_and_drafts_messages(self) -> Iterator[str]:
+        """Get the messages of all commits and open drafts.
 
         Yields:
-            The latest commit.
+            The commit or draft message.
 
         """
+        latest_draft: Optional[Draft] = None
         while True:
             try:
-                latest_commit = self._sorted_commits[-1].pop(0)
+                latest_commit = self._sorted_logs[-1][0]
             except IndexError:
+                if latest_draft:
+                    yield self._draft_printer(latest_draft)
+                if self._sorted_drafts:
+                    yield from map(self._draft_printer, reversed(self._sorted_drafts))
                 return
-            if self._merge(latest_commit):
-                continue
-            yield latest_commit
-            self._sort()
-
-    def generate_commits(self) -> Iterator[str]:
-        """Get the the log of all commits.
+            if not latest_draft and self._sorted_drafts:
+                latest_draft = self._sorted_drafts.pop()
+            if latest_draft and latest_draft.updated_at > latest_commit.committer.date:
+                yield self._draft_printer(latest_draft)
+                latest_draft = None
+            else:
+                del self._sorted_logs[-1][0]
+                if self._merge(latest_commit):
+                    continue
+                yield self._commit_printer(
+                    latest_commit, self._commit_id_to_branches.get(latest_commit.commit_id)
+                )
+                self._sort()
+
+
+class _RootCommitNode:  # pylint: disable=too-many-instance-attributes
+    SIGN = "*"
+
+    def __init__(self) -> None:
+        self.parent: Optional[_RootCommitNode] = None
+        self.date = 0
+        self.key = ROOT_COMMIT_ID
+        self.available_child_num = 0
+        self.get_oneline_message: Callable[
+            [List[str]], str
+        ] = lambda _: _ONELINE_COMMIT_MESSAGE.format(shorten(ROOT_COMMIT_ID), "ROOT_COMMIT")
+        self.get_full_message: Callable[
+            [List[str]], str
+        ] = lambda _: f'{click.style("ROOT_COMMIT {}", fg="yellow")}'.format(self.key)
```

Comment:
Is it possible to make them class methods?

Reply:
It will cause mypy raising error at `_CommitNode`'s initialization